src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsassessforearlyreleaseapi/entity/AssessmentStatus.kt

```diff
@@ -285,16 +138,436 @@ enum class AssessmentStatus {
+    }
+    on<AssessmentLifecycleEvent.FailEligibilityAndSuitability> {
+      transitionTo(AssessmentState.IneligibleOrUnsuitable)
+    }
+    on<AssessmentLifecycleEvent.Postpone> {
+      transitionTo(AssessmentState.Postponed)
+    }
+    on<AssessmentLifecycleEvent.Timeout> {
+      transitionTo(AssessmentState.TimedOut)
+    }
+    on<AssessmentLifecycleEvent.OptOut> {
+      transitionTo(AssessmentState.OptedOut(this))
+    }
+  }
+
+  state<AssessmentState.AddressUnsuitable> {
+    on<AssessmentLifecycleEvent.CompleteEligibilityAndSuitability> {
+      transitionTo(AssessmentState.EligibleAndSuitable)
+    }
+    on<AssessmentLifecycleEvent.SubmitForAddressChecks> {
+      transitionTo(AssessmentState.AwaitingRefusal)
+    }
+    on<AssessmentLifecycleEvent.FailEligibilityAndSuitability> {
+      transitionTo(AssessmentState.IneligibleOrUnsuitable)
+    }
+    on<AssessmentLifecycleEvent.Timeout> {
+      transitionTo(AssessmentState.TimedOut)
+    }
+    on<AssessmentLifecycleEvent.OptOut> {
+      transitionTo(AssessmentState.OptedOut(this))
+    }
+  }
+
+  state<AssessmentState.AwaitingRefusal> {
+    on<AssessmentLifecycleEvent.Timeout> {
+      transitionTo(AssessmentState.TimedOut)
+    }
+    on<AssessmentLifecycleEvent.Refuse> {
+      transitionTo(AssessmentState.Refused)
+    }
+    on<AssessmentLifecycleEvent.CompleteEligibilityAndSuitability> {
+      transitionTo(AssessmentState.EligibleAndSuitable)
+    }
+    on<AssessmentLifecycleEvent.SubmitForAddressChecks> {
+      transitionTo(AssessmentState.AwaitingAddressAndRiskChecks)
+    }
+  }
+
+  state<AssessmentState.IneligibleOrUnsuitable> {
+    on<AssessmentLifecycleEvent.StartEligibilityAndSuitability> {
+      transitionTo(AssessmentState.EligibilityAndSuitabilityInProgress)
+    }
+    on<AssessmentLifecycleEvent.CompleteEligibilityAndSuitability> {
+      transitionTo(AssessmentState.EligibleAndSuitable)
+    }
+    on<AssessmentLifecycleEvent.SubmitForAddressChecks> {
+      transitionTo(AssessmentState.AwaitingAddressAndRiskChecks)
+    }
+    on<AssessmentLifecycleEvent.StartAddressChecks> {
+      transitionTo(AssessmentState.AddressAndRiskChecksInProgress)
+    }
+    on<AssessmentLifecycleEvent.SubmitForDecision> {
+      transitionTo(AssessmentState.AwaitingPreDecisionChecks)
+    }
+    on<AssessmentLifecycleEvent.Approve> {
+      transitionTo(AssessmentState.Approved)
+    }
+    on<AssessmentLifecycleEvent.CompleteAddressChecks> {
+      transitionTo(AssessmentState.AwaitingPreReleaseChecks)
+    }
+    on<AssessmentLifecycleEvent.ReleaseOnHDC> {
+      transitionTo(AssessmentState.PassedPreReleaseChecks)
+    }
+    on<AssessmentLifecycleEvent.FailAddressChecks> {
+      transitionTo(AssessmentState.AddressUnsuitable)
+    }
+    on<AssessmentLifecycleEvent.Refuse> {
+      transitionTo(AssessmentState.Refused)
+    }
+    on<AssessmentLifecycleEvent.OptOut> {
+      transitionTo(AssessmentState.OptedOut(this))
```

Comment:
so if `Ineligible/Unsuitable`, if the user selects to `optOut` then move to `OptedOut`and store the previous stage as `Ineligible/Unsuitable`